packages/lib-classifier/src/plugins/tasks/survey/components/SurveyTask.js

```diff
@@ -1,24 +1,91 @@
 import { Box } from 'grommet'
+import { observer } from 'mobx-react'
 import PropTypes from 'prop-types'
-import React from 'react'
+import React, { useEffect } from 'react'
 
 import Choice from './components/Choice'
 import Chooser from './components/Chooser'
 
-function SurveyTask({
-  answers = {},
+export function SurveyTaskContainer ({
+  annotation,
+  task,
   disabled = false,
-  filters = {},
-  previousChoice = '',
-  handleAnswers = () => {},
-  handleChoice = () => {},
-  handleDelete = () => {},
-  handleFilter = () => {},
-  handleIdentify = () => {},
-  selectedChoice = '',
-  selectedChoiceIds = [],
-  task
 }) {
+  const [answers, setAnswers] = React.useState({})
+  const [filters, setFilters] = React.useState({})
+  const [previousChoice, setPreviousChoice] = React.useState('')
+  const [selectedChoice, setSelectedChoice] = React.useState('')
+
+  useEffect(() => {
+    return () => {
+      if (annotation._choiceInProgress) {
+        annotation.setChoiceInProgress(false)
+      }
+    }
+  })
```

Comment:
This sets `annotation._choiceInProgress` to false on every change of props or state, if it's true. You could replace it with the following if block, but I'm not really sure what this block is doing.

```suggestion
  if (annotation._choiceInProgress) {
    annotation.setChoiceInProgress(false)
  }
```

Reply:
It seems like `annotation._choiceInProgress` can never be true. What is it used for?

Reply:
Great catch! `choiceInProgress` is used to prevent clicking Done (and related) while a choice is selected. Here's [a link to the survey task model](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/plugins/tasks/survey/models/SurveyTask.js#L58) that uses `choiceInProgress` within the `isComplete` view.

I just added a commit with a related failing test and a commit to edit the survey task component useEffect with a dependency on the task prop to fix the failing test.

Reply:
Thanks. That makes sense.